### PR TITLE
Fix 3178

### PR DESCRIPTION
**Pull request by kzisme**
# Pull Request Checklist

Resolves: #3178  

- [X] Ensured **tests pass** and (if applicable) updated functional test output
- [X] Conformed to **code style guidelines** by running appropriate linting tools
- [ ] Added **tests** for changed code
- [ ] Updated **documentation** for changed code

I wasn't able to get the standard unit tests working locally, but it's a very small change.  

I attempted to run the unit tests described [here](https://github.com/getpelican/pelican/wiki/Git-Tips#making-your-changes) but I think I was missing some sort of initial setup.  

I was able to re-produce this bug via the tumblr API (although I ran into another error when the tool was going through the specific blog in this issue - which was related to expecting a str literal but found a bool).  

[here](https://gist.github.com/kzisme/87d1592cd914be62306475b8678d9dd4) is the API response from tumblr for the specific post in question along with the command I ran to reproduce, image link, and blog link.

### CODE REVIEW

File: pelican/tools/pelican_import.py [hunk 1]
```diff
@@ -468,9 +468,8 @@ def get_tumblr_posts(api_key, blogname, offset=0):
                     fmtstr = '<img alt="%s" src="%s" />'
                 content = ''
                 for photo in post.get('photos'):
-                    content += '\n'.join(
-                        fmtstr % (photo.get('caption'),
-                                  photo.get('original_size').get('url')))
+                    content += fmtstr % (photo.get('caption'),
+                                  photo.get('original_size').get('url'))
```
Comment (avaris) [inline pelican/tools/pelican_import.py:472]: I think `'\n'.join(...)` was to keep one image per line if there are multiple and I'd say it's better to keep it in place for nicer output. But the implementation was wrong. So, the whole thing, including `for`, probably should be something like:

```python
content += '\n'.join(fmtstr % (photo.get('caption'),
                               photo.get('original_size').get('url'))
                     for photo in post.get('photos'))
```

PS: Also, you need to lineup your indentation properly to make linter happy.

Comment (kzisme) [reply pelican/tools/pelican_import.py]: Ahh, that makes sense - I only tested with one image.  Sorry for the slight delay in getting back to this!

I had/have Pylint setup on my end - is there a different linter that is used for Pelican?  If so, can you point me in the direction to get it working properly?  I think I resolved everything in your comment, and have it indented properly, but let me know if there is anything else glaring.

Comment (justinmayer) [reply pelican/tools/pelican_import.py]: At the moment, Pelican uses Flake8. You can see the two linter failures in the CI output: https://github.com/getpelican/pelican/pull/3187/checks#step:5:20

The dev environment setup docs are here: https://docs.getpelican.com/en/latest/contribute.html#setting-up-the-development-environment

File: pelican/tools/pelican_import.py [hunk 2]
```diff
@@ -468,10 +468,9 @@ def get_tumblr_posts(api_key, blogname, offset=0):
                     fmtstr = '<img alt="%s" src="%s" />'
                 content = ''
                 for photo in post.get('photos'):
-                    content += '\n'.join(
-                        fmtstr % (photo.get('caption'),
-                                  photo.get('original_size').get('url')))
-                content += '\n\n' + post.get('caption')
+                    content += '\n'.join(fmtstr % (photo.get('caption'),
+                                                   photo.get('original_size').get('url'))
+                                        for photo in post.get('photos'))
```
Comment (boxydog) [inline pelican/tools/pelican_import.py:473]: Is this right? It looks like you're iterating across post.get('photos') on line 473, but it's already doing that on line 470.

**Comment by boxydog** [comment]
I would close this, as equivalent functionality was merged in #3215.

**Comment by justinmayer** [comment]
Thanks, Kevin, for the contribution, the equivalent of which has now been merged. 🌟

**Pull request closed without merging**